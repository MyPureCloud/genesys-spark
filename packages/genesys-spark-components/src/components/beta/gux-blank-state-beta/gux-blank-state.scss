@use '~genesys-spark/dist/beta/tokens/flare/ui.scss';

$gse-ui-blankState-large-minWidth: 570px; // --gse-ui-blankState-large-minWidth

:host {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  container-type: inline-size;
}

.gux-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: ui.$gse-ui-blankState-padding;
  color: ui.$gse-ui-blankState-foregroundColor;

  &.gux-no-padding {
    padding: 0;
  }

  .gux-call-to-actions {
    display: flex;
    flex-direction: column;
    gap: 4px; // Token Required : COMUI-4166
    padding-block-start: ui.$gse-ui-blankState-gapContent;
  }

  .gux-image {
    padding-block-end: ui.$gse-ui-blankState-gapMain;
  }

  .gux-message {
    padding-block-end: ui.$gse-ui-blankState-gapMessage;
  }

  ::slotted(gux-icon) {
    inline-size: ui.$gse-ui-blankState-icon-size-lg;
    block-size: ui.$gse-ui-blankState-icon-size-lg;
    color: ui.$gse-ui-blankState-iconColor;
  }

  slot[name='primary-message'] {
    font-family: ui.$gse-ui-progressAndLoading-blankState-large-header-fontFamily;
    font-size: ui.$gse-ui-progressAndLoading-blankState-large-header-fontSize;
    font-weight: ui.$gse-ui-progressAndLoading-blankState-large-header-fontWeight;
    line-height: ui.$gse-ui-progressAndLoading-blankState-large-header-lineHeight;
    color: ui.$gse-ui-blankState-foregroundColor;
    text-align: center;
  }

  slot[name='additional-guidance'] {
    font-family: ui.$gse-ui-progressAndLoading-blankState-large-subheader-fontFamily;
    font-size: ui.$gse-ui-progressAndLoading-blankState-large-subheader-fontSize;
    font-weight: ui.$gse-ui-progressAndLoading-blankState-large-subheader-fontWeight;
    line-height: ui.$gse-ui-progressAndLoading-blankState-large-subheader-lineHeight;
    color: ui.$gse-ui-blankState-foregroundColor;
    text-align: center;
  }

  &.gux-alignment-left {
    align-items: flex-start;

    slot[name='primary-message'],
    slot[name='additional-guidance'] {
      text-align: start;
    }
  }
}

// https://bholmes.dev/blog/alternative-to-css-variable-media-queries/
@container (width < #{$gse-ui-blankState-large-minWidth}) {
  .gux-container {
    ::slotted(gux-icon) {
      inline-size: ui.$gse-ui-blankState-icon-size-sm;
      block-size: ui.$gse-ui-blankState-icon-size-sm;
    }

    slot[name='primary-message'] {
      font-family: ui.$gse-ui-progressAndLoading-blankState-small-header-fontFamily;
      font-size: ui.$gse-ui-progressAndLoading-blankState-small-header-fontSize;
      font-weight: ui.$gse-ui-progressAndLoading-blankState-small-header-fontWeight;
      line-height: ui.$gse-ui-progressAndLoading-blankState-small-header-lineHeight;
    }

    slot[name='additional-guidance'] {
      font-family: ui.$gse-ui-progressAndLoading-blankState-small-subheader-fontFamily;
      font-size: ui.$gse-ui-progressAndLoading-blankState-small-subheader-fontSize;
      font-weight: ui.$gse-ui-progressAndLoading-blankState-small-subheader-fontWeight;
      line-height: ui.$gse-ui-progressAndLoading-blankState-small-subheader-lineHeight;
    }
  }
}
