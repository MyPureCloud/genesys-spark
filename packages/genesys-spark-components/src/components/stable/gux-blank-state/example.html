<h1>gux-blank-state</h1>

<h4>
  The blank state will transition from large to medium or small when the view
  becomes smaller. Use the the drag icon to see the difference.
</h4>

<gux-blank-state class="blank-state-styling">
  <gux-icon slot="image" icon-name="bot" decorative="true"></gux-icon>
  <div slot="primary-message">Sorry, something went wrong.</div>
  <div slot="additional-guidance">Please refresh this page to try again.</div>
  <button slot="call-to-action" type="button" onclick="notify(event)">
    Call to action
  </button>
</gux-blank-state>

<style>
  .blank-state-styling {
    inline-size: 300px;
    block-size: 250px;
    overflow: auto;
    resize: horizontal;
    border: 1px dashed gainsboro;
  }
</style>
