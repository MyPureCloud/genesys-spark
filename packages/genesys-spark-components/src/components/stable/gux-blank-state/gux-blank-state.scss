@use '../../../../../genesys-spark-tokens/dist/scss/_gse-ui-feedback'; // TODO: https://inindca.atlassian.net/browse/COMUI-2242

:host {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  container-type: inline-size;
}

.gux-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: var(--gse-ui-blankState-padding);
  color: var(--gse-ui-blankState-foregroundColor);
  background-color: var(--gse-ui-blankState-backgroundColor);
  border-radius: var(--gse-ui-blankState-borderRadius);

  .gux-image {
    padding-bottom: var(--gse-ui-blankState-gapMain);
  }

  .gux-message {
    padding-bottom: var(--gse-ui-blankState-gapMessage);
  }

  .gux-guidance {
    padding-bottom: var(--gse-ui-blankState-gapContent);
  }

  ::slotted(gux-icon) {
    width: var(--gse-ui-blankState-icon-size-lg);
    height: var(--gse-ui-blankState-icon-size-lg);
    color: var(--gse-ui-blankState-iconColor);
  }

  slot[name='primary-message'] {
    font-family: var(
      --gse-ui-progressAndLoading-blankState-large-header-fontFamily
    );
    font-size: var(
      --gse-ui-progressAndLoading-blankState-large-header-fontSize
    );
    font-weight: var(
      --gse-ui-progressAndLoading-blankState-large-header-fontWeight
    );
    line-height: var(
      --gse-ui-progressAndLoading-blankState-large-header-lineHeight
    );
    color: var(--gse-ui-blankState-foregroundColor);
    text-align: center;
  }

  slot[name='additional-guidance'] {
    font-family: var(
      --gse-ui-progressAndLoading-blankState-large-subheader-fontFamily
    );
    font-size: var(
      --gse-ui-progressAndLoading-blankState-large-subheader-fontSize
    );
    font-weight: var(
      --gse-ui-progressAndLoading-blankState-large-subheader-fontWeight
    );
    line-height: var(
      --gse-ui-progressAndLoading-blankState-large-subheader-lineHeight
    );
    color: var(--gse-ui-blankState-foregroundColor);
    text-align: center;
  }
}

// https://bholmes.dev/blog/alternative-to-css-variable-media-queries/
@container (width < #{gse-ui-feedback.$gse-ui-blankState-large-minWidth}) {
  .gux-container {
    ::slotted(gux-icon) {
      width: var(--gse-ui-blankState-icon-size-sm);
      height: var(--gse-ui-blankState-icon-size-sm);
    }

    slot[name='primary-message'] {
      font-family: var(
        --gse-ui-progressAndLoading-blankState-small-header-fontFamily
      );
      font-size: var(
        --gse-ui-progressAndLoading-blankState-small-header-fontSize
      );
      font-weight: var(
        --gse-ui-progressAndLoading-blankState-small-header-fontWeight
      );
      line-height: var(
        --gse-ui-progressAndLoading-blankState-small-header-lineHeight
      );
    }

    slot[name='additional-guidance'] {
      font-family: var(
        --gse-ui-progressAndLoading-blankState-small-subheader-fontFamily
      );
      font-size: var(
        --gse-ui-progressAndLoading-blankState-small-subheader-fontSize
      );
      font-weight: var(
        --gse-ui-progressAndLoading-blankState-small-subheader-fontWeight
      );
      line-height: var(
        --gse-ui-progressAndLoading-blankState-small-subheader-lineHeight
      );
    }
  }
}
