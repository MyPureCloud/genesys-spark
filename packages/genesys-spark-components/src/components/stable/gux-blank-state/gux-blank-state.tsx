import { Component, Element, JSX, h, State } from '@stencil/core';
import { trackComponent } from '@utils/tracking/usage';
import { OnMutation } from '@utils/decorator/on-mutation';
import { hasSlot } from '@utils/dom/has-slot';

/**
 * @slot primary-message - Required slot for primary-message.
 * @slot image - Slot for gux-icon element.
 * @slot additional-guidance - Slot for additional-guidance.
 * @slot call-to-action - Slot for the message call to action button.
 */

@Component({
  styleUrl: 'gux-blank-state.scss',
  tag: 'gux-blank-state',
  shadow: true
})
export class GuxBlankState {
  @Element()
  root: HTMLElement;

  @State()
  private hasCallToAction: boolean = false;

  @OnMutation({ childList: true, subtree: true })
  onMutation(): void {
    this.hasCallToAction = hasSlot(this.root, 'call-to-action');
  }

  private renderCallToActionSlot(): JSX.Element {
    if (this.hasCallToAction) {
      return (
        <gux-button-slot accent="primary">
          <slot name="call-to-action"></slot>
        </gux-button-slot>
      ) as JSX.Element;
    }
  }

  componentWillLoad() {
    trackComponent(this.root);

    this.hasCallToAction = hasSlot(this.root, 'call-to-action');
  }

  render(): JSX.Element {
    return (
      <div class="gux-container">
        <div class="gux-image">
          <slot name="image"></slot>
        </div>
        <div class="gux-message">
          <slot name="primary-message"></slot>
        </div>
        <div class="gux-guidance">
          <slot name="additional-guidance"></slot>
        </div>
        {this.renderCallToActionSlot()}
      </div>
    ) as JSX.Element;
  }
}
