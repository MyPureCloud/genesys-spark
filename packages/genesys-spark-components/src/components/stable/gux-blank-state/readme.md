# gux-blank-state

<!-- Auto Generated Below -->


## Slots

| Slot                    | Description                                 |
| ----------------------- | ------------------------------------------- |
| `"additional-guidance"` | Slot for additional-guidance.               |
| `"call-to-action"`      | Slot for the message call to action button. |
| `"image"`               | Slot for gux-icon element.                  |
| `"primary-message"`     | Required slot for primary-message.          |


## Dependencies

### Depends on

- [gux-button-slot](../gux-button-slot)

### Graph
```mermaid
graph TD;
  gux-blank-state --> gux-button-slot
  style gux-blank-state fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
