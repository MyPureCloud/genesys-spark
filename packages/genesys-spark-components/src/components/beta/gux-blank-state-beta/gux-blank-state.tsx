import { Component, Element, JSX, h, State, Prop } from '@stencil/core';
import { trackComponent } from '@utils/tracking/usage';
import { OnMutation } from '@utils/decorator/on-mutation';
import { hasSlot } from '@utils/dom/has-slot';
import { GuxBlankStateAlignment } from './gux-blank-state.types';

/**
 * @slot primary-message - Required slot for primary-message.
 * @slot image - Slot for gux-icon element or gux-illustration-beta.
 * @slot additional-guidance - Slot for additional-guidance.
 * @slot call-to-action - Slot for the message call to action button.
 * @slot secondary-call-to-action - Slot for the message call to action secondary button.
 */

@Component({
  styleUrl: 'gux-blank-state.scss',
  tag: 'gux-blank-state-beta',
  shadow: true
})
export class GuxBlankStateBeta {
  @Element()
  root: HTMLElement;

  @State()
  private hasCallToAction: boolean = false;

  @State()
  private hasSecondaryCallToAction: boolean = false;

  @State()
  private hasGuidance: boolean = false;

  @Prop()
  alignment: GuxBlankStateAlignment = 'center';

  @Prop()
  noPadding: boolean = false;

  @OnMutation({ childList: true, subtree: true })
  onMutation(): void {
    this.hasCallToAction = hasSlot(this.root, 'call-to-action');
    this.hasGuidance = hasSlot(this.root, 'additional-guidance');
    this.hasSecondaryCallToAction = hasSlot(
      this.root,
      'secondary-call-to-action'
    );
  }

  private renderImageSlot(): JSX.Element {
    return (<slot name="image"></slot>) as JSX.Element;
  }

  private renderCallToActions(): JSX.Element {
    if (this.hasCallToAction) {
      return (
        <div class="gux-call-to-actions">
          {this.hasCallToAction && (
            <gux-button-slot accent="primary">
              <slot name="call-to-action"></slot>
            </gux-button-slot>
          )}
          {this.hasSecondaryCallToAction && (
            <gux-button-slot accent="ghost">
              <slot name="secondary-call-to-action"></slot>
            </gux-button-slot>
          )}
        </div>
      ) as JSX.Element;
    }
  }

  private renderGuidanceSlot(): JSX.Element {
    if (this.hasGuidance) {
      return (
        <div class="gux-guidance">
          <gux-truncate maxLines={3}>
            <slot name="additional-guidance"></slot>
          </gux-truncate>
        </div>
      ) as JSX.Element;
    }
  }

  componentWillLoad() {
    trackComponent(this.root);

    this.hasCallToAction = hasSlot(this.root, 'call-to-action');
    this.hasGuidance = hasSlot(this.root, 'additional-guidance');
    this.hasSecondaryCallToAction = hasSlot(
      this.root,
      'secondary-call-to-action'
    );
  }

  render(): JSX.Element {
    return (
      <div
        class={{
          'gux-container': true,
          'gux-no-padding': this.noPadding,
          'gux-alignment-left': this.alignment === 'left'
        }}
      >
        <div class="gux-image">{this.renderImageSlot()}</div>
        <div class="gux-message">
          <slot name="primary-message"></slot>
        </div>
        {this.renderGuidanceSlot()}
        {this.renderCallToActions()}
      </div>
    ) as JSX.Element;
  }
}
