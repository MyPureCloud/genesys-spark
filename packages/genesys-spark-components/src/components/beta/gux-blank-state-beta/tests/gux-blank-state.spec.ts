import { GuxBlankStateBeta } from '../gux-blank-state';
import { checkRenders } from '@test/specTestUtils';
import { renderConfigs } from './gux-blank-state.common';

const components = [GuxBlankStateBeta];

describe('#render', () => {
  checkRenders(renderConfigs, components);
});
