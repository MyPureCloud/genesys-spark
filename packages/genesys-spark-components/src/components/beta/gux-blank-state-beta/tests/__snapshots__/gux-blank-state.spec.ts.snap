// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`#render should render a gux-blank-state component 1`] = `
<gux-blank-state-beta>
  <template shadowrootmode="open">
    <div class="gux-container">
      <div class="gux-image">
        <slot name="image">
        </slot>
      </div>
      <div class="gux-message">
        <slot name="primary-message">
        </slot>
      </div>
      <div class="gux-guidance">
        <gux-truncate maxlines="3">
          <slot name="additional-guidance">
          </slot>
        </gux-truncate>
      </div>
      <div class="gux-call-to-actions">
        <gux-button-slot accent="primary">
          <slot name="call-to-action">
          </slot>
        </gux-button-slot>
      </div>
    </div>
  </template>
  <gux-icon slot="image" icon-name="bot" decorative="true">
  </gux-icon>
  <div slot="primary-message">
    Sorry, something went wrong.
  </div>
  <div slot="additional-guidance">
    Please refresh this page to try again.
  </div>
  <button slot="call-to-action" type="button">
    Call to action
  </button>
</gux-blank-state-beta>
`;

exports[`#render should render component as expected when "additional-guidance" slot is provided 1`] = `
<gux-blank-state-beta>
  <template shadowrootmode="open">
    <div class="gux-container">
      <div class="gux-image">
        <slot name="image">
        </slot>
      </div>
      <div class="gux-message">
        <slot name="primary-message">
        </slot>
      </div>
      <div class="gux-guidance">
        <gux-truncate maxlines="3">
          <slot name="additional-guidance">
          </slot>
        </gux-truncate>
      </div>
      <div class="gux-call-to-actions">
        <gux-button-slot accent="primary">
          <slot name="call-to-action">
          </slot>
        </gux-button-slot>
      </div>
    </div>
  </template>
  <div slot="primary-message">
    Sorry, something went wrong.
  </div>
  <button slot="call-to-action" type="button">
    Call to action
  </button>
  <div slot="additional-guidance">
    Please refresh this page to try again.
  </div>
</gux-blank-state-beta>
`;

exports[`#render should render component as expected when "image" slot is provided 1`] = `
<gux-blank-state-beta>
  <template shadowrootmode="open">
    <div class="gux-container">
      <div class="gux-image">
        <slot name="image">
        </slot>
      </div>
      <div class="gux-message">
        <slot name="primary-message">
        </slot>
      </div>
      <div class="gux-call-to-actions">
        <gux-button-slot accent="primary">
          <slot name="call-to-action">
          </slot>
        </gux-button-slot>
      </div>
    </div>
  </template>
  <div slot="primary-message">
    Sorry, something went wrong.
  </div>
  <button slot="call-to-action" type="button">
    Call to action
  </button>
  <gux-illustration-beta slot="image" variant="success" status="success">
  </gux-illustration-beta>
</gux-blank-state-beta>
`;

exports[`#render should render component as expected when "secondary-call-to-action" slot is provided 1`] = `
<gux-blank-state-beta>
  <template shadowrootmode="open">
    <div class="gux-container">
      <div class="gux-image">
        <slot name="image">
        </slot>
      </div>
      <div class="gux-message">
        <slot name="primary-message">
        </slot>
      </div>
      <div class="gux-call-to-actions">
        <gux-button-slot accent="primary">
          <slot name="call-to-action">
          </slot>
        </gux-button-slot>
        <gux-button-slot accent="ghost">
          <slot name="secondary-call-to-action">
          </slot>
        </gux-button-slot>
      </div>
    </div>
  </template>
  <div slot="primary-message">
    Sorry, something went wrong.
  </div>
  <button slot="call-to-action" type="button">
    Call to action
  </button>
  <button slot="secondary-call-to-action" type="button">
    Call to action
  </button>
</gux-blank-state-beta>
`;

exports[`#render should render component with alignment "center" 1`] = `
<gux-blank-state-beta alignment="center">
  <template shadowrootmode="open">
    <div class="gux-container">
      <div class="gux-image">
        <slot name="image">
        </slot>
      </div>
      <div class="gux-message">
        <slot name="primary-message">
        </slot>
      </div>
    </div>
  </template>
  <gux-icon slot="image" icon-name="bot" decorative="true">
  </gux-icon>
  <div slot="primary-message">
    center aligned blank state
  </div>
</gux-blank-state-beta>
`;

exports[`#render should render component with alignment "left" 1`] = `
<gux-blank-state-beta alignment="left">
  <template shadowrootmode="open">
    <div class="gux-container gux-alignment-left">
      <div class="gux-image">
        <slot name="image">
        </slot>
      </div>
      <div class="gux-message">
        <slot name="primary-message">
        </slot>
      </div>
    </div>
  </template>
  <gux-icon slot="image" icon-name="bot" decorative="true">
  </gux-icon>
  <div slot="primary-message">
    left aligned blank state
  </div>
</gux-blank-state-beta>
`;

exports[`#render should render component with no-padding attribute 1`] = `
<gux-blank-state-beta no-padding="">
  <template shadowrootmode="open">
    <div class="gux-container gux-no-padding">
      <div class="gux-image">
        <slot name="image">
        </slot>
      </div>
      <div class="gux-message">
        <slot name="primary-message">
        </slot>
      </div>
    </div>
  </template>
  <gux-icon slot="image" icon-name="bot" decorative="true">
  </gux-icon>
  <div slot="primary-message">
    No padding blank state
  </div>
</gux-blank-state-beta>
`;
